### PR TITLE
Inspect epochs with state-viewer

### PR DESCRIPTION
**Pull request by nikurt**
Currently there is no way to directly inspect EpochInfos in the RocksDB.
This PR lets state-viewer make scans and queries to the EpochInfos column, and also lets us inspect blocks and chunks produced and expected by a given validator (#5688)

Will provide an example output of the tool soon.

### CODE REVIEW

File: test-utils/state-viewer/src/epoch_info.rs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    epoch_manager: &mut EpochManager,
+    runtime_adapter: Arc<dyn RuntimeAdapter>,
+) {
+    println!("{:?}: {:#?}", epoch_id, epoch_info);
+    if epoch_info.epoch_height() >= *head_epoch_height {
+        println!("Epoch information for this epoch is not yet available, skipping.");
+        return;
+    }
+    if let Some(account_id) = validator_account_id.clone() {
+        if let Some(kickout) = epoch_info.validator_kickout().get(&account_id) {
```
Comment (mm-near) [inline test-utils/state-viewer/src/epoch_info.rs:167]: optional: I'd refactor 167- 212 into a separate method (that displays info from specific validator point of view)

File: test-utils/state-viewer/src/epoch_info.rs [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+                    let block = chain_store.get_block(&block_hash).unwrap();
+                    if block.chunks()[shard_id as usize].height_included() != block_height {
+                        missing_chunks.push((block_height, shard_id));
+                    }
+                } else {
+                    missing_chunks.push((block_height, shard_id));
+                }
+            }
+            println!("Missing {} chunks: {:?}", missing_chunks.len(), missing_chunks);
+        }
```
Comment (mm-near) [inline test-utils/state-viewer/src/epoch_info.rs:211]: else --> print("validator didn't validate in this epoch")

**Review by mm-near — APPROVED** [review]

**Review by pmnoxx — APPROVED** [review]
LGTM

File: test-utils/state-viewer/src/cli.rs [hunk 3]
```diff
@@ -243,3 +247,40 @@ impl DumpAccountStorageCmd {
         );
     }
 }
+#[derive(Clap)]
+pub struct EpochInfoCmd {
+    /// Fetch the given epoch.
```
Comment (pmnoxx) [inline test-utils/state-viewer/src/cli.rs:252]: I looked at `get_epoch_ids`, it looks like those options are exclusive.

Even if you specify `epoch_id`, `epoch_height` option will be ignored.
It look to me like, all those options should belong to a group and be mutually exlusive.

File: test-utils/state-viewer/src/commands.rs [hunk 4]
```diff
@@ -490,6 +491,45 @@ pub(crate) fn check_block_chunk_existence(store: Arc<Store>, near_config: NearCo
     }
     println!("Block check succeed");
```
Comment (pmnoxx) [inline test-utils/state-viewer/src/commands.rs:492]: Although, this is not part of this PR. It may be  a good idea to switch to using tracing, rather than !println!

File: test-utils/state-viewer/src/epoch_info.rs [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+use near_store::{DBCol, Store};
+use std::sync::Arc;
+
+pub(crate) fn print_epoch_info(
+    epoch_id: Option<EpochId>,
+    epoch_height: Option<EpochHeight>,
+    block_hash: Option<CryptoHash>,
+    block_height: Option<BlockHeight>,
+    protocol_version: Option<ProtocolVersion>,
+    validator_account_id: Option<AccountId>,
```
Comment (pmnoxx) [inline test-utils/state-viewer/src/epoch_info.rs:19]: maybe, should be enum, when I read `get_epoch_ids`, I realized, it only makes sense for one of those options to be specified.
